### PR TITLE
Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by makkes**
**What type of PR is this?**
Documentation

**What this PR does/ why we need it**:
Added a CONTRIBUTING file with instructions for potential contributors. This is to clarify what addon maintainers have to do in order to get their PRs merged as quickly as possible.

**Which issue(s) this PR fixes**:
no issue

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Checklist**

* [x] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [x] The documentation is updated where needed.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+1. Create a PR that removes all files that are no longer supported
+2. Once the PR merges, a new minor release of this repository should be created indicating removal of support for the older versions
+
+Once this is complete end users who are using older releases of `kubernetes-base-addons` will be unaffected.
+
+## Addon Revisions
+
+You will find that any particular addon directory (e.g. `addons/prometheus`) may have several directories and several manifests nested in them each with variants of that addon. These are what we refered to above as "revisions".
+
+The **intention of revisions is to maintain a flat history of addon changes**. If you are making changes to any particular addon you should be making a revision of that addon as a copy of the original file with the changes made therein and the `addon-revision` version updated to reflect the new version appropriately.
```
Comment (hectorj2f) [inline CONTRIBUTING.md]: Well, I think this paragraph should clarify when to create a new directory and when create a new file inside of the existing directory

Comment (makkes) [reply CONTRIBUTING.md]: True. @shaneutt are you able to come up with a sentence or two?

Comment (shaneutt) [reply CONTRIBUTING.md]:
```suggestion
The **intention of revisions is to maintain a flat history of addon changes**. If you are making changes to any particular addon you should be making a revision of that addon as a copy of the original file with the changes made therein and the `addon-revision` version updated to reflect the new version appropriately.

New directories can be created for minor versions of a release (e.g. a directory named `v0.9.x`) and contain any revisions matching that minor release. You create new files which should follow the patter `<addon_name>-<version>-<revision>` (e.g. `helloworld-v0.9.1-1.yaml`) keeping in mind revisions start over from 1 again if the patch version updates.
```

Comment (shaneutt) [reply CONTRIBUTING.md]: I think we can start with :point_up_2: to just have something for now, and expand the language as needed.

**Pull request merged**